### PR TITLE
[FLINK-29385] Verify duplicate column names for AddColumn

### PR DESCRIPTION
**Pull request by zjureel**
When we execute AddColumn ddl, we should check whether the column name is exist. Otherwise, the ddl will be successful and the job reads this table will create source of flink store table failed.

### CODE REVIEW

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java [hunk 1]
```diff
@@ -173,6 +174,15 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
                     newOptions.remove(removeOption.key());
                 } else if (change instanceof AddColumn) {
                     AddColumn addColumn = (AddColumn) change;
+                    Set<String> fieldNames =
```
Comment (SteNicholas) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java:177]: Why not directly filter the field name of add column in the stream of `newFields`?

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java [hunk 2]
```diff
@@ -173,6 +174,15 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
...
+                            newFields.stream().map(DataField::name).collect(Collectors.toSet());
+                    if (fieldNames.contains(addColumn.fieldName())) {
+                        throw new IllegalArgumentException(
+                                "The column["
```
Comment (SteNicholas) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java:181]:
```suggestion
                                String.format("The column [%s] exists in the table[%s].".format(addColumn.fieldName(), tableRoot))
```

File: flink-table-store-core/src/test/java/org/apache/flink/table/store/table/SchemaEvolutionTest.java [hunk 3]
```diff
@@ -120,6 +122,28 @@ public void testAddField() throws Exception {
...
+        schemaManager.commitChanges(
+                Collections.singletonList(SchemaChange.addColumn("f3", new BigIntType())));
+        assertThatThrownBy(
+                        () -> {
+                            schemaManager.commitChanges(
+                                    Collections.singletonList(
+                                            SchemaChange.addColumn("f3", new FloatType())));
+                        })
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageStartingWith("The column[f3] is exist in");
```
Comment (SteNicholas) [inline flink-table-store-core/src/test/java/org/apache/flink/table/store/table/SchemaEvolutionTest.java:144]: Why not use `hasMessage` to verify the complete exception message？

**Review by SteNicholas — CHANGES_REQUESTED** [review]
Thanks for @zjureel contribution. I left some comments on this pull request. PTAL.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java (hunk 1 above)
Comment (JingsongLi) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java:177]: `Stream.anyMatch`

Comment (zjureel) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java]: > Why not directly filter the field name of add column in the stream of `newFields`?

It's an interesting question. Maybe that the data types of these two columns are different, so i think it's better to return the error message to the user

**Comment by zjureel** [comment]
Thanks @JingsongLi @SteNicholas  I have updated the codes

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**